pylint/checkers/base.py

```diff
@@ -1882,6 +1902,28 @@ def _raise_name_warning(self, node, node_type, name, confidence):
         self.add_message("invalid-name", node=node, args=args, confidence=confidence)
         self.stats["badname_" + node_type] += 1
 
+    def _name_valid_due_to_whitelist(self, name: str) -> bool:
+        good_names_patterns = self.config.good_names_rgxs
+        matches = [
+            match
+            for match in (pattern.search(name) for pattern in good_names_patterns)
```

Comment:
I think we should compile these exceptions for better performance. We can do that in the `open()` method, where we have access to `self.config`.

On top of that, I'd rewrite this function as:

```suggestion
         name in self.config.good_names or any(pattern.match(name) for pattern in self.compiled_patterns)
```

Reply:
done